ddtrace/internal/nogevent.py

```diff
@@ -37,7 +37,30 @@ def is_module_patched(module):
 thread_get_ident = get_original(six.moves._thread.__name__, "get_ident")
 Thread = get_original("threading", "Thread")
 Lock = get_original("threading", "Lock")
-RLock = get_original("threading", "RLock")
+
+if six.PY2 and is_module_patched("threading"):
+    _allocate_lock = get_original("threading", "_allocate_lock")
+    _threading_RLock = get_original("threading", "_RLock")
+
+    class _RLock(_threading_RLock):
+        """Patched RLock to ensure threading._allocate_lock is called rather than
+        gevent.threading._allocate_lock if patching has occurred. This is not
+        necessary in Python 3 where the RLock function uses the _CRLock so is
+        unaffected by gevent patching.
+        """
+
+        def __init__(self):
+            self.__block = _allocate_lock()
+            self.__owner = None
+            self.__count = 0
```

Comment:
```suggestion
        def __init__(self):
            super(_RLock, self).__init__()
            self.__block = _allocate_lock()
```

?

Reply:
I wanted to avoid the unnecessary creationg of a gevent lock. wdyt?

Reply:
yeah, I am not sure personally... the likelihood of the API/properties changing is very minimal, so yeah, this is probably perfectly fine